Sources/Rendering/Misc/TextureLODsDownloader/index.js

```diff
@@ -0,0 +1,184 @@
+import macro from 'vtk.js/Sources/macro';
+
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+
+const getRemoteFileSize = (url) => {
```

Comment:
This function obtains the size of a remote file by checking the `Content-Length` header after making a `HEAD` request. For some reason, though, firefox is not returning a `Content-Length` header for cross origin requests. It seems to work fine for non-cross origin requests, though. But I am not sure why it doesn't work, nor what I could do to fix it, and I also do not know of another way to get the remote file size. Chrome seems to work fine in all cases I tried.

Reply:
Did you try without expecting readyState === 4 and status == 200 ?
What do you see in the network tab with firefox?

Reply:
Yes, I tried that, but unfortunately, it does not help.

Strangely, in the network tab with firefox there is a `content-length` for the xhr CORS HEAD requests, which makes me think it must be present somewhere. But it does not show up in javascript, and I do not know why. If I print out `xhr.getAllResponseHeaders()` in firefox, I get something like the following:

```
cache-control: max-age=600
content-type: image/jpeg
expires: Tue, 31 Dec 2019 21:03:44 GMT
last-modified: Tue, 31 Dec 2019 02:34:55 GMT
```

You can see the `content-length` is not there. However, if the request has the same origin, the `content-length` IS there. And this is what `xhr.getAllResponseHeaders()` looks like in chrome, for the CORS HEAD request:

```
cache-control: max-age=600
content-length: 73223341
content-type: image/jpeg
expires: Tue, 31 Dec 2019 21:03:50 GMT
last-modified: Tue, 31 Dec 2019 02:34:55 GMT
```

Almost exactly the same, but the `content-length` is present.

I really do not know why it isn't working in firefox. But it makes it so that this function will not work for html files that we email around.

Reply:
Firefox currently doesn't support accessing Content-Length headers from CORS responses, even though it's part of the spec: https://fetch.spec.whatwg.org/#cors-safelisted-response-header-name

I'll see if I can get a fix into Firefox, since I know where the fix should be.

Reply:
@floryst Sounds good! Unfortunately, since I don't know of any work-arounds for this, we might just have to leave it as is for now, and wait until the issue is fixed. Until that time, standalone html files will not work with firefox with this function.